Add dashboard guidelines (#6818)
DataDog/integrations-core: docs/developer/guidelines/dashboards.md

# Dashboards

-----

Datadog dashboards enable you to efficiently monitor your infrastructure and integrations
 by displaying and tracking key metrics.
 
## Integration Preset Dashboards

If you would like to create a default dashboard for an integration, follow the guidelines in the [Best Practices](./dashboards.md#best-practices) section.

### Exporting a dashboard payload

> When you've created a dashboard, you can export the dashboard payload to be included in the integration assets.

Reviewer: Maybe this could be something like
```suggestion
When you've created a dashboard in the Datadog UI, you can export the dashboard payload to be included in its integration's assets directory.
```

Run the following command to [export the dashboard](../ddev/cli.md#export):

```cli

> ddev meta dash export <URL_OF_DASHBOARD> <INTEGRATION>

Reviewer 2: Maybe we should tell they need to configure the keys before being able to do that (https://datadoghq.dev/integrations-core/ddev/cli/#set)

```

!!! tip
    If the dashboard is for a contributor-maintained integration in the `integration-extras` repo, run the command with the `--extras` or `-e` flag.

The command will add the dashboard definition to the `manifest.json` file of the integration. 
The dashboard JSON payload will be available in `/assets/dashboards/<INTEGRATION>_overview.json`.

Commit the changes and create a pull request.

## Best Practices

![Mongo dashboard](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/mongo_dashboard.png)

1. When creating a new dashboard, select the Screenboard type.

    !!! information
        Integration preset (or out of the box) dashboards are typically screenboards because they allow images, widgets, and labels. 
        Learn more about Datadog dashboards and see the differences between [screenboards and timeboards](https://docs.datadoghq.com/dashboards/#screenboard-vs-timeboard). 

1. Research the metrics supported by the integration and consider grouping them in relevant categories. 
Important metrics that are key to the performance and overview of the integration should be at the top.

> 1. Ensure good design by keeping labels and widgets aligned with consistent spacing and reasonable size. 

Reviewer 2: Maybe suggest checking the size with the TV mode as in the best practice document

> Make the whole dashboard fit on your horizontal screen (check it with the TV mode).

### Logos

Integration logos should be placed at the top-left of the dashboard and centered on the [image widget](https://docs.datadoghq.com/dashboards/widgets/image/).

The suggested height for logos is around 12 grid units.

### Labels

Labels are created from the [Notes & Links](https://docs.datadoghq.com/dashboards/widgets/note/) widget type.

#### Metric Categories
Labels are used to divide the screenboard in visually comprehensive sections. 

- Category labels have gray backgrounds to contrast with the board and widgets.
- Text is 18px and horizontally centered. 

    !!! important
        Do not use bold or all capitalized letters because it can make the dashboard harder to read.

> - Horizontal labels should be kept above the graphs and its width should be the length of the entire section. 

Reviewer: ```suggestion
- Horizontal labels should be kept above the graphs and their width should be the length of the entire section. 
```

- For vertical labels, keep the label to the left of the section and its length should be the height the graphs. 

> Enable the right pointer to properly convey the direction of the section. 

Reviewer: right could be confusing (right vs left)
```suggestion
Enable a pointer to properly convey the direction of the section. 
```

# Dashboards

-----

Datadog dashboards enable you to efficiently monitor your infrastructure and integrations
 by displaying and tracking key metrics.
 
## Integration Preset Dashboards

If you would like to create a default dashboard for an integration, follow the guidelines in the [Best Practices](./dashboards.md#best-practices) section.

### Exporting a dashboard payload
When you've created a dashboard in the Datadog UI, you can export the dashboard payload to be included in its integration's assets directory.

Ensure that you have set an `API_KEY` and `APP_KEY` for the org that contains the new dashboard in the ddev [configuration](https://datadoghq.dev/integrations-core/ddev/cli/#set).

Run the following command to [export the dashboard](../ddev/cli.md#export):

```cli
ddev meta dash export <URL_OF_DASHBOARD> <INTEGRATION>
```

!!! tip
    If the dashboard is for a contributor-maintained integration in the `integration-extras` repo, run the command with the `--extras` or `-e` flag.

The command will add the dashboard definition to the `manifest.json` file of the integration. 

> The dashboard JSON payload will be available in `/assets/dashboards/<INTEGRATION>_overview.json`.

Reviewer: The filename is the lowercase name of the original dashboard

Once your PR is merged and synced on production, you can find your dashboard in the Dashboard List page.

!!! tip
    Make sure the integration tile is `Installed` in order to see the preset dashboard in the list.

Ensure logos render correctly on the Dashboard List page and within the preset dashboard. 

## Best Practices

![Mongo dashboard](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/mongo_dashboard.png)

1. When creating a new dashboard, select the Screenboard type.

    !!! information
        Integration preset (or out of the box) dashboards are typically screenboards because they allow images, widgets, and labels. 
        Learn more about Datadog dashboards and see the differences between [screenboards and timeboards](https://docs.datadoghq.com/dashboards/#screenboard-vs-timeboard). 

1. Research the metrics supported by the integration and consider grouping them in relevant categories. 
Important metrics that are key to the performance and overview of the integration should be at the top.

1. Ensure good design by keeping labels and widgets aligned with consistent spacing and reasonable size. 

> 

Reviewer: Could we also include naming conventions here?

The command will add the dashboard definition to the `manifest.json` file of the integration. 
The dashboard JSON payload will be available in `/assets/dashboards/<DASHBOARD_TITLE>.json`.

Commit the changes and create a pull request.

### Verify the Preset Dashboard

Once your PR is merged and synced on production, you can find your dashboard in the Dashboard List page.

!!! tip
    Make sure the integration tile is `Installed` in order to see the preset dashboard in the list.

Ensure logos render correctly on the Dashboard List page and within the preset dashboard. 

## Best Practices

![Mongo dashboard](https://raw.githubusercontent.com/DataDog/integrations-core/master/docs/developer/assets/images/mongo_dashboard.png)

1. When creating a new dashboard, select the Screenboard type.

    !!! information
        Integration preset (or out of the box) dashboards are typically screenboards because they allow images, widgets, and labels. 
        Learn more about Datadog dashboards and see the differences between [screenboards and timeboards](https://docs.datadoghq.com/dashboards/#screenboard-vs-timeboard). 

> 1. Dashboard titles should contain the integration name. Some examples of a good dashboard title is `Syclla` or `Cilium Overview`.

Reviewer: ```suggestion
1. Dashboard titles should contain the integration name. Some examples of a good dashboard title are `Syclla` and `Cilium Overview`.
```
